android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java

```diff
@@ -66,6 +68,62 @@ public void setMarkdownStyle(@NonNull MarkdownStyle markdownStyle) {
     mMarkdownStyle = markdownStyle;
   }
 
+  private List<MarkdownRange> splitRangesOnEmojis(List<MarkdownRange> markdownRanges) {
+    List<MarkdownRange> emojiRanges = new ArrayList<>();
+    for (MarkdownRange range : markdownRanges) {
+      if (range.type.equals("emoji")) {
+        emojiRanges.add(range);
+      }
+    }
+
+    for (int i = 0; i < markdownRanges.size(); i++) {
+      MarkdownRange range = markdownRanges.get(i);
+      if (!range.type.equals("italic") && !range.type.equals("strikethrough")) {
+        continue;
+      }
+
+      for (int j = 0; j <  emojiRanges.size(); j++) {
+        MarkdownRange emojiRange = emojiRanges.get(j);
+        if (emojiRange.start >= range.start && emojiRange.end <= range.end) {
+          // Split range
+          MarkdownRange startRange = new MarkdownRange(range.type, range.start, emojiRange.start, range.depth);
+          MarkdownRange endRange =  new MarkdownRange(range.type, emojiRange.end, range.end, range.depth);
+          markdownRanges.add(i + 1, endRange);
+          markdownRanges.add(i + 1, startRange);
```

Comment:
`markdownRanges` is an `ArrayList`, wouldn't it be better to use a `LinkedList` if we want to insert elements in the middle of the list?

Reply:
Back then, LinkedList might be a better idea, but with the current enhanced O(n) algorithm, we need ArrayList to replace elements in an array of all ranges